spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -106,6 +109,45 @@ public Mqttv5PahoMessageDrivenChannelAdapter(MqttConnectionOptions connectionOpt
 		}
 	}
 
+	/**
+	 * Use this constructor when you need to use a single {@link ClientManager}
+	 * (for instance, to reuse an MQTT connection).
+	 * @param clientManager The client manager.
+	 * @param topic The topic(s).
+	 */
+	public Mqttv5PahoMessageDrivenChannelAdapter(ClientManager<IMqttAsyncClient> clientManager, String... topic) {
+		this(buildDefaultConnectionOptions(null), clientManager, topic);
+	}
+
+	/**
+	 * Use this constructor when you need to use a single {@link ClientManager}
+	 * (for instance, to reuse an MQTT connection) and a specific {@link MqttConnectionOptions}.
+	 * @param connectionOptions The connection options.
+	 * @param clientManager The client manager.
+	 * @param topic The topic(s).
+	 */
+	public Mqttv5PahoMessageDrivenChannelAdapter(MqttConnectionOptions connectionOptions,
```

Comment:
I believe we discussed previously with you that this ctor is wrong.
If we really need options internally, then `ClientManager` must expose its options.
See `MqttComponent` to be extended from this `ClientManager`.

Reply:
`MqttComponent` interface requires two methods implementation: `setBeanName` and `getBeanName` - do you have ideas what is the better way to implement them? All current impl classes just extend `IntegrationObjectSupport` and this helps. However we agreed to have more lightweight client managers, so probably not the best way to extend it...

Reply:
See `BeanNameAware`: just have an impl in the `AbstractMqttClientManager` and store `beanName` to be available from the `getBeanName()`.
The `BeanNameAware` is the one of `Aware` which is called by Spring inversion of control container.

Reply:
Ok, will have it as a private field, then. Thanks for pointing it.